plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java

```diff
@@ -325,9 +328,17 @@ else if (isRowType(type)) {
                 BlockBuilder builder = output.beginBlockEntry();
 
                 checkState(type.getTypeParameters().size() == 3, "DBRef should have 3 fields : %s", type);
-                appendTo(type.getTypeParameters().get(0), dbRefValue.getDatabaseName(), builder);
-                appendTo(type.getTypeParameters().get(1), dbRefValue.getCollectionName(), builder);
-                appendTo(type.getTypeParameters().get(2), dbRefValue.getId(), builder);
+                for (int i = 0; i < type.getTypeSignature().getParameters().size(); i++) {
```

Comment:
Any specific reason for using `TypeSignature` here for fetching parameter details when we could use `RowType` directly - is it to decouple the DBRef and Trino type representation ?